tests/e2e/functional_test.go

```diff
@@ -524,6 +541,54 @@ var _ = Describe("Functional e2e", Serial, func() {
 
 	})
 
+	It("Should update the child ISBService with an excluded field", func() {
+
+		// new ISBService spec
+		rawSpec, err := json.Marshal(ISBServiceSpecExcludedField)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		updateISBServiceRolloutInK8S(isbServiceRolloutName, func(rollout apiv1.ISBServiceRollout) (apiv1.ISBServiceRollout, error) {
+			rollout.Spec.InterStepBufferService.Spec.Raw = rawSpec
+			return rollout, nil
+		})
+
+		if dataLossPrevention == "true" {
+
+			document("Verify that in-progress-strategy gets set to NoOp")
+			verifyInProgressStrategyISBService(Namespace, isbServiceRolloutName, apiv1.UpgradeStrategyNoOp)
+			verifyInProgressStrategy(pipelineRolloutName, apiv1.UpgradeStrategyNoOp)
+
+			document("Verify that dependent Pipeline is not paused when an update to ISBService not requiring pause is made")
+			verifyPipelineStatusConsistently(Namespace, pipelineName, func(retrievedPipelineSpec numaflowv1.PipelineSpec, retrievedPipelineStatus numaflowv1.PipelineStatus) bool {
+				return retrievedPipelineStatus.Phase != numaflowv1.PipelinePhasePaused
+			})
+
+			Eventually(func() bool {
+				isbRollout, _ := isbServiceRolloutClient.Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
+				isbCondStatus := getRolloutCondition(isbRollout.Status.Conditions, apiv1.ConditionPausingPipelines)
+				plRollout, _ := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
+				plCondStatus := getRolloutCondition(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
+				if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
+					return false
+				}
+				return true
+			}, testTimeout).Should(BeTrue())
```

Comment:
We really need to do this "consistently", not "eventually", right? We kind of need to verify that lines 562-575 are happening consistently.

Reply:
That sounds right, how long should we check - 30 seconds? 1 minute?

Reply:
How about 20 sec or something around there?

Reply:
So it looks like consistently will immediately fail as soon as the pausing pipelines condition is true unlike eventually which waits till they're both false. That's why it keeps failing quickly.

Reply:
Taking a look at the logs.. the Pipeline does pause. That shouldn't be happening in this case, hm.

Reply:
> So it looks like consistently will immediately fail as soon as the pausing pipelines condition is true unlike eventually which waits till they're both false

Yes, when we check `Consistently`, we do need to check `Eventually` first so we can make sure it got to that state in the first place

Reply:
Ideally we can first "Eventually" and then "Consistently" check:

- verifyInProgressStrategyISBService(Namespace, isbServiceRolloutName, apiv1.UpgradeStrategyNoOp)
- verifyInProgressStrategy(pipelineRolloutName, apiv1.UpgradeStrategyNoOp)
- PipelineStatus != Pausing or Paused
- PausingConditions false

To avoid repeating a large chunk of code, if we can't think of good functions, we can always create a temporary function to call each time within this function like:
```
verifyNotPausing := func(str string) {
       .....
    }

Consistently(....
   verifyNotPausing
}

Eventually(...
   verifyNotPausing
}
```